crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -1668,10 +1666,27 @@ fn generate_tangents_for_mesh(mesh: &Mesh) -> Result<Vec<[f32; 4]>, GenerateTang
     Ok(mikktspace_mesh.tangents)
 }
 
+/// Correctly scales and renormalizes an already normalized `normal` by the scale determined by its reciprocal `scale_recip`
+fn scale_normal(normal: Vec3, scale_recip: Vec3) -> Vec3 {
+    // This is basically just `normal * scale_recip` but with the added rule that `0. * anything == 0.`
+    // This is necessary because one of the components of `scale_recip` is infinite
+    let n = Vec3::select(normal.cmpeq(Vec3::ZERO), Vec3::ZERO, normal * scale_recip);
+
+    // If n is finite, the normal was perpendicular to the flattened axis
+    // else the
+    if n.is_finite() {
+        n.normalize_or_zero()
+    } else {
+        Vec3::select(n.abs().cmpeq(Vec3::INFINITY), n.signum(), Vec3::ZERO)
```

Comment:
Note this might not be normalized when there's several axes that are infinities
```suggestion
        Vec3::select(n.abs().cmpeq(Vec3::INFINITY), n.signum(), Vec3::ZERO).normalize()
```

Reply:
You are right, but that can't really happen because the scaling of meshes is not allowed when two axes of `scale` are zero / when two axes of `scale_recip` are infinite